clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy

```diff
@@ -75,30 +91,233 @@ class AzureLoadBalancerCachingAgent implements CachingAgent, AccountAware {
   }
 
   @Override
+  String getOnDemandAgentType() {
+    "${getAgentType()}-OnDemand"
+  }
+
+  @Override
   Collection<AgentDataType> getProvidedDataTypes() {
     return types
   }
 
   @Override
+  boolean handles(String type) {
+    type == LEGACY_ON_DEMAND_TYPE
+  }
+
+  @Override
+  boolean handles(String type, String cloudProvider) {
+    type == ON_DEMAND_TYPE && cloudProvider == azureCloudProvider.id
+  }
+
+  @Override
+  OnDemandAgent.OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("loadBalancerName")) {
+      return null
+    }
+    if (!data.containsKey("account")) {
+      return null
+    }
+    if (!data.containsKey("region")) {
+      return null
+    }
+
+    if (accountName != data.account) {
+      return null
+    }
+
+    if (region != data.region) {
+      return null
+    }
+
+    AzureLoadBalancerDescription updatedLoadBalancer = null
+    AzureLoadBalancerDescription evictedLoadBalancer = null
+    String loadBalancerName = data.loadBalancerName as String
+    String resourceGroupName = AzureUtilities.getResourceGroupName(AzureUtilities.getAppNameFromAzureResourceName(loadBalancerName), region)
+    if (resourceGroupName == null) {
+      log.info("handle->Unexpected error retrieving resource group name: null")
+      return []
+    }
+
+    try {
+      updatedLoadBalancer = metricsSupport.readData {
+        creds.networkClient.getLoadBalancer(creds, resourceGroupName, loadBalancerName)
+      }
+    } catch (Exception e ) {
+      log.info("handle->Unexpected exception: ${e.message}")
+      return null
+    }
+
+    def cacheResult = metricsSupport.transformData {
+      if (updatedLoadBalancer) {
+        return buildCacheResult(providerCache, null, 0, updatedLoadBalancer, null)
+      } else {
+        evictedLoadBalancer = new AzureLoadBalancerDescription(
+          loadBalancerName: loadBalancerName,
+          region: region,
+          appName: AzureUtilities.getAppNameFromAzureResourceName(loadBalancerName),
+          cloudProvider: "azure",
+          cluster: "none",
+          lastReadTime: System.currentTimeMillis()
+        )
+        return buildCacheResult(providerCache, null, 0, null, evictedLoadBalancer)
+      }
+    }
+    Map<String, Collection<String>> evictions = evictedLoadBalancer ? [(Keys.Namespace.AZURE_LOAD_BALANCERS.ns): [getLoadBalancerKey(evictedLoadBalancer)]] : [:]
+
+    log.info("onDemand cache refresh (data: ${data}, evictions: ${evictions})")
+    return new OnDemandAgent.OnDemandResult(
+      sourceAgentType: getAgentType(), cacheResult: cacheResult, evictions: evictions
+    )
+  }
+
+  @Override
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    return []
+  }
```

Comment:
Ah, ok. Maybe we should look to see if it's ever implemented and look to clean up the base class.  Worth checking maybe? Not a big deal either way though.